modules/command-engine/core/src/main/scala/surge/internal/core/SurgePartitionRouterImpl.scala

```diff
@@ -60,11 +66,17 @@ private[surge] final class SurgePartitionRouterImpl(
       businessLogic.partitioner,
       businessLogic.kafka.stateTopic,
       regionCreator,
-      RoutableMessage.extractEntityId)(businessLogic.tracer)
+      RoutableMessage.extractEntityId,
+      kafkaProducerOverride)(businessLogic.tracer)
 
     val routerActorName = s"${businessLogic.aggregateName}RouterActor"
-    val shardRouter = system.actorOf(shardRouterProps, name = routerActorName)
-    shardRouter
+    val shardRouterManaged = ActorLifecycleManagerActor.manage(
```

Comment:
Let's leave this not wrapped in a managed actor for now if possible. Given the subtlety of the bug around this that was fixed and the fact that the router actor goes away when we swap to using Akka Cluster I'm a bit anxious that the tradeoff of being able to control the actor in this way isn't worth the risk of some other subtle change coming along and completely breaking multi-node down the line

Reply:
Ok, agree with you. Thanks! I'll remove it completely